src/components/ModularTable/ModularTable.tsx

```diff
@@ -53,14 +59,23 @@ function ModularTable({
           return (
             <TableRow {...row.getRowProps()}>
               {row.cells.map((cell) => {
+                const hasColumnIcon = cell.column.getCellIcon;
+                const iconName = hasColumnIcon && cell.column.getCellIcon(cell);
+
                 return (
                   <TableCell
                     {...cell.getCellProps([
                       {
                         className: cell.column.className,
                       },
+                      {
+                        className: hasColumnIcon
+                          ? "p-table__cell--icon-placeholder"
+                          : "",
+                      },
                     ])}
                   >
+                    {iconName && <Icon name={iconName} />}
```

Comment:
@squidsoup 

I tried the approach with defining two separate props and separate types in TS like this:

<img width="1304" alt="Screenshot 2020-09-24 at 12 01 18" src="https://user-images.githubusercontent.com/83575/94131297-0b687f00-fe5e-11ea-9091-07100760a2bd.png">

It worked on its own, correctly suggested types for name, etc.

But it felt short in this line, because having separate props in `Icon` component, means any components that use the `Icon` need to make this distinction as well, which becomes problematic if we want them to easily and consistently support both built-in and custom icons.

So I hope it's good enough as is for now.

Reply:
Another approach would be to provide a typescript enum for the default icons.

```ts
export enum ICONS {
  plus = "plus,
  minus = "minus",
  ...
}
```

Which would be used like this:

```jsx
import { Icon, ICONS } from "@canonical/react-components"
 
<Icon name={ICONS.plus} />
<Icon name="custom-icon" />
```

Reply:
Great suggestion @huwshimi, thanks!